OrbitQt/orbitmainwindow.cpp

```diff
@@ -1165,11 +1173,38 @@ void OrbitMainWindow::SetTarget(const orbit_qt::StadiaTarget& target) {
   app_->SetGrpcChannel(connection->GetGrpcChannel());
   app_->SetProcessManager(target.GetProcessManager());
   app_->SetTargetProcess(target.GetProcess());
+
   // TODO (b/173011168): Color should not be defined inline. Replace with defined "good color"
   // (green).
-  target_label_->setStyleSheet("#TargetLabel { color: #66BB6A; }");
+  static const QString kTargetLabelDefaultStyleSheet = "#TargetLabel { color: #66BB6A; }";
+  target_label_->setStyleSheet(kTargetLabelDefaultStyleSheet);
   target_label_->setText(QString::fromStdString(target.GetProcess()->name()) + " @ " +
                          target.GetConnection()->GetInstance().display_name);
+
+  using ProcessInfo = orbit_grpc_protos::ProcessInfo;
+  target.GetProcessManager()->SetProcessListUpdateListener([&](std::vector<ProcessInfo> processes) {
+    // This lambda is called from a background-thread, so we use QMetaObject::invokeMethod
+    // to execute our logic on the main thread.
+    QMetaObject::invokeMethod(this, [&, processes = std::move(processes)]() {
+      const auto is_current_process = [this](const auto& process) {
+        const auto target_process = app_->GetTargetProcess();
+        return target_process != nullptr && process.pid() == app_->GetTargetProcess()->pid();
+      };
+      const auto current_process =
+          std::find_if(processes.begin(), processes.end(), is_current_process);
+
+      if (const auto process_died = current_process == processes.end(); process_died) {
```

Comment:
I don't see the benefit of the `process_died` assignment here since it's not used inside the if-block. Is it purely to convey intent of `current_process == processes.end()`? For me, it's more confusing then helping because I've never used this if-syntax in the past, but maybe that's just me.

In any case, `auto` instead of `bool` is certainly not helping ;-)

Reply:
Yes, it's merely to explain that `current_process == processes.end()` means that the process died. But I can write that differently.

Reply:
Changed!